model/licensing.py

```diff
@@ -32,11 +27,76 @@ class PolicyException(Exception):
     pass
 
 
-class License(Base):
+class LicenseStatus(PythonEnum):
+    preorder = "preorder"
+    available = "available"
+    unavailable = "unavailable"
+
+    @classmethod
+    def get(cls, value: str):
+        return cls.__members__.get(value.lower(), cls.unavailable)
+
+
+class LicenseFunctions:
+    identifier: str
+    checkout_url: str
+    status_url: str
+    status: LicenseStatus
+    expires: Optional[datetime.datetime]
+    checkouts_left: Optional[int]
+    checkouts_available: int
+    terms_concurrency: Optional[int]
+
+    @property
+    def is_perpetual(self) -> bool:
+        return (self.expires is None) and (self.checkouts_left is None)
+
+    @property
+    def is_time_limited(self) -> bool:
+        return self.expires is not None
+
+    @property
+    def is_loan_limited(self) -> bool:
+        return self.checkouts_left is not None
+
+    @property
+    def is_expired(self) -> bool:
+        now = utc_now()
+        return (
+            (self.expires and self.expires <= now)
+            or (self.checkouts_left is not None and self.checkouts_left <= 0)
+            or (self.status != LicenseStatus.available)
+        )
+
+    @property
+    def owned(self) -> int:
```

Comment:
The term "owned" is driving me bonkers (not a long drive, I realize 😄). It's more like a point-in-time "total loans remaining" (which, in some cases might have the same value forever) vs. available, which is kind of "loans available right now." It would be nice to have property names that make this more clear.

Reply:
I went with `owned` and `available` to match the terms used in `LicensePool`, since these properties are used to calculate these values on the pool [here](https://github.com/ThePalaceProject/circulation-core/pull/36/files#diff-bb2d07cab580f09523b6335301dc1e0fedde8a70f162afa6152ec14c289172b1R661-R662). I'm up for changing them though, if you have suggestions for changes to them.

Reply:
How about `total_remaining_loans` and `currently_available_loans` (though, I don't have as much of a concern about `available` as I do about `owned`). I mostly interested in not perpetuating the confusing property names from `licensepools`.

Reply:
Updated to use those property names.